src/procstat.c

```diff
@@ -215,27 +215,27 @@ static bool root_directory(struct procstat_context *context, struct procstat_dir
 	return &context->root == directory;
 }
 
-__attribute__((__nonnull__(1,2))) static void procstat_lock(struct procstat_context *context, struct procstat_directory *item) {
+__attribute__((__nonnull__(1))) static void procstat_lock(struct procstat_directory *item) {
 	assert(item != NULL);
 
 	if (item->mutexptr) {
 		pthread_mutex_lock(item->mutexptr);
 	}
 	else {
 		assert(item->parent);
-		procstat_lock(context, item->base.parent);
+		procstat_lock(item->base.parent);
```

Comment:
IMHO this takes us back to where we crashed last week: there's no guarantee that item->parent is valid.

Reply:
Sorry, please, disregard this comment - github did not show the up to date commits.